### PR TITLE
Provider: add `linear-docs` provider

### PR DESCRIPTION
**Pull request by valerybugakov**
- Adds the Linear Docs provider, which is very similar to the Linear Issues provider
- Added instructions for internal testing to 1Password

## Test plan

1. Tested using the local `bundle.js` file.
2. Tested using the published: https://www.npmjs.com/package/@openctx/provider-linear-docs

### CODE REVIEW

File: provider/linear-docs/auth.ts [hunk 1]
```diff
@@ -0,0 +1,111 @@
+// Identical to provider/linear-issues/auth.ts.
```
Comment (toolmantim) [inline provider/linear-docs/auth.ts:1]: If this is identical, should we just created a shared lib file for them, or import one from the other?

Comment (valerybugakov) [reply provider/linear-docs/auth.ts:1]: I thought about it but decided to keep it simple for now with code duplication. I do not expect this code (local auth scripts) to be changed. WDYT?

Comment (toolmantim) [reply provider/linear-docs/auth.ts:1]: It's a fair whack of code… but we don't have CI that runs across both if you just intend on changing one… so probably not a bad idea to keep separate.

**Review by toolmantim — APPROVED** [review]
Works great!

CleanShot 2024-06-06 at 15 53 42@2x

**Pull request merged**